src/freedv_api.c

```diff
@@ -1269,8 +1269,7 @@ void freedv_set_clip(struct freedv *f, int val) {
 /* Band Pass Filter to cleanup OFDM tx waveform, only supported by some modes */
 
 void freedv_set_tx_bpf(struct freedv *f, int val) {
-    if (FDV_MODE_ACTIVE( FREEDV_MODE_700D, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_700E, f->mode) 
-        || FDV_MODE_ACTIVE( FREEDV_MODE_DATAC0, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_DATAC3, f->mode)) {
+    if (is_ofdm_mode(f)) {
         ofdm_set_tx_bpf(f->ofdm, val);
```

Comment:
Yes I think this is the correct approach, if indeed compression is set up for all OFDM modes (voice and data).  We may be able to go a little further;
1. Look for other long lists of modes in the compression code and replace with `is_ofdm_mode(f)`.
2. Make sure there are asserts to bomb out if we hit a mode that doesn't have compression set up (like forgetting to define a filter).

I'll poke about some more over the next few days (or you are welcome to take a look).

Reply:
Go for it 👍